### DISCUSSION TITLE
How does one specify an anchor at the origin?

### DISCUSSION THREAD

**Discussion by ToddRAllen** [BelfrySCAD/BOSL2#1431] [Q&A] [answered]
Using a previous version of BOSL2 I built a model which was a union of multiple calls of the skin() module with each being passed a sequence of profiles generated by offset_stroke().  The profiles were aligned in X & Y by the origin and the lists of Z values passed to each call of skin() aligned the resulting polyhedrons in Z.  Now that skin() aligns to anchors I don't know how to preserve my desired alignment.  Is there a way to specify an anchor at the origin?

**Comment by adrianVmariano** [comment, marked as answer]
Anchoring to the origin should be achieved with anchor="origin", which is the default.  This appears to be working properly.  The docs seem to be missing the section on named anchors for skin().

**Comment by ToddRAllen** [comment]
Thanks!  I had faiiled to see what had changed is a need  to add a parameter to offset_stroke() of cp=[0,0] to get the same behavior as I was getting before.  Here is an example illustrating the issue and fix:

include <BOSL2/std.scad>
include <BOSL2/rounding.scad>
include <BOSL2/skin.scad>

p = [[3,3],[13,3]];
color("red") offset_stroke(p);
color("green") skin([offset_stroke(p), offset_stroke(p)], z=[0.5,1.5], slices=1);
color("blue") skin([offset_stroke(p,cp=[0,0]), offset_stroke(p,cp=[0,0])], z=[1.5,2.5], slices=1);
![skinning-offset_stroke](https://github.com/BelfrySCAD/BOSL2/assets/17037725/ae9c7198-f723-4d82-817b-706c2cbe4d33)

I did not expect the difference between offset_stroke(p) called as a module seen in red versus when called as a function and skinned as seen in green.

**Comment by adrianVmariano** [comment]
That is a bug in offset stroke.  It is supposed to have a default anchor of "origin" but the default isn't given.  Setting cp is one workaround.   Should be fixed in the PR.
